lib/features/children/edit_child/pages/edit_child_page.dart

```diff
@@ -0,0 +1,234 @@
+import 'dart:developer';
+
+import 'package:currency_text_input_formatter/currency_text_input_formatter.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:givt_app/core/enums/amplitude_events.dart';
+import 'package:givt_app/core/enums/country.dart';
+import 'package:givt_app/features/auth/cubit/auth_cubit.dart';
+import 'package:givt_app/features/children/create_child/widgets/create_child_text_field.dart';
+import 'package:givt_app/features/children/details/cubit/child_details_cubit.dart';
+import 'package:givt_app/features/children/details/models/profile_ext.dart';
+import 'package:givt_app/features/children/edit_child/cubit/edit_child_cubit.dart';
+import 'package:givt_app/features/children/edit_child/models/edit_child.dart';
+import 'package:givt_app/features/children/edit_child/widgets/giving_allowance_info_button.dart';
+import 'package:givt_app/features/children/utils/child_date_utils.dart';
+import 'package:givt_app/l10n/l10n.dart';
+import 'package:givt_app/utils/utils.dart';
+import 'package:go_router/go_router.dart';
+import 'package:intl/intl.dart';
+
+class EditChildPage extends StatefulWidget {
+  const EditChildPage({super.key});
+
+  @override
+  State<EditChildPage> createState() => _EditChildPageState();
+}
+
+class _EditChildPageState extends State<EditChildPage> {
+  final _nameController = TextEditingController();
+  final _dateOfBirthController = TextEditingController();
+  final _allowanceController = TextEditingController();
+
+  void _editChild() {
+    final name = _nameController.text.trim();
+
+    final allowance = _allowanceController.text.isNotEmpty
+        ? int.parse(
+            _allowanceController.text
+                .trim()
+                .substring(1), // removing currency sign
+          )
+        : 0;
+
+    final child = EditChild(
+      firstName: name,
+      allowance: allowance,
+    );
+    context.read<EditChildCubit>().editChild(child: child);
+  }
+
+  void _updateInputFields(
+    ProfileExt profileDetails,
+    EditChild child,
+    String currencySymbol,
+  ) {
+    _nameController.text = child.firstName;
+    _dateOfBirthController.text = ChildDateUtils.dateFormatter.format(
+      DateTime.parse(profileDetails.dateOfBirth),
+    );
+    final allowanceText =
+        '$currencySymbol${child.allowance.toStringAsFixed(0)}';
+    _allowanceController.text = allowanceText;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final locals = AppLocalizations.of(context);
+
+    final user = context.read<AuthCubit>().state.user;
+    final currency = NumberFormat.simpleCurrency(
+      name: Util.getCurrencyName(country: Country.fromCode(user.country)),
+    );
+
+    return BlocConsumer<EditChildCubit, EditChildState>(
```

Comment:
Same, you don't need this top level consumer and you can nest it into the body for instace and display from there the snackbars according to the state

Reply:
same here, I need a state object on an AppBar level